### PR TITLE
LT-238 Add reject function to pixelz gem.

### PR DESCRIPTION
**Pull request by monogot**

### CODE REVIEW

**Comment by n-thongjor** [comment]
@gunosk129 You are the first reviewer!, please review. Type :+1: when you've done.

File: app/models/pixelz/modification_request.rb [hunk 1]
```diff
@@ -19,7 +19,20 @@ def self.raise_pixelz_error(res)
...
+      res = Pixelz::ApiClient::Image.new(modifiable).reject(pixelz_ticket, comment)
+      raise_reject_pixelz_error(res)
+      res
+    end
+    
     private_class_method :raise_pixelz_error
 
+    private
+
+    def raise_reject_pixelz_error(res)
```
Comment (ekkapob) [inline app/models/pixelz/modification_request.rb:33]: Would it be better if we create a share method to raise error for above `self.raise_pixelz_error` function and `raise_reject_pixelz_error` ? 

Both of them are private and differ only the keys`AddImageResponse` and `RejectImageResponse`.

```
def self.modify(image) 
  raise_pixelz_error(res, 'AddImageResponse')
  ...
end

def reject(comment)
  raise_pixelz_error(res, 'RejectImageResponse')
  ...
end

private
def raise_pixelz_error(res, key)
  return if res[key]['ErrorCode'] == 'NoError'
  raise Pixelz::Error.new res[key]['Message']
end
```

File: lib/pixelz/api_client/image.rb [hunk 2]
```diff
@@ -12,24 +12,40 @@ def post
         Hash.from_xml(res)
       end
 
+      def reject(image_ticket, comment='')
+        res = RestClient.put(BASE_URI + "/Image/Reject/#{image_ticket}", reject_image_payload(comment), headers)
```
Comment (ekkapob) [inline lib/pixelz/api_client/image.rb:16]: The line is long, should make it new line. If you use sublime, there is ruler option below, make it 80. If vim, need to ask Paul.

http://stackoverflow.com/questions/25900954/80-characters-right-margin-line-in-sublime-text-3

File: spec/fixtures/fake_pixelz_api_responses/image_reject.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<RejectImageResponse >
```
Comment (ekkapob) [inline spec/fixtures/fake_pixelz_api_responses/image_reject.xml:1]: Shouldn't it be <RejectImageResponse> (no space) ? Also the Closing tag.

File: spec/models/pixelz/modification_request_spec.rb [hunk 4]
```diff
@@ -46,4 +46,25 @@
...
   end
+
+  describe '#reject' do
+    before do
+      @image = create(:image)
+      @modify_image_request = Pixelz::ModificationRequest.modify(@image) 
+    end
+    context 'with comment' do
+      it 'returns no error code' do
+        comment = "The top of picture still have background colour."
```
Comment (ekkapob) [inline spec/models/pixelz/modification_request_spec.rb:57]: User single quote whenever possible if there is no string interpolation. This helps the entire app consistency. Below codes as well, `res["RejectImageResponse"]["ErrorCode"] , comment = "", etc`

**Pull request closed without merging**